app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,80 @@
+require 'pry'
+
+class TweetsController < ApplicationController
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :'/tweets/new'
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      if @tweet.user == current_user
+        erb :'/tweets/edit'
+      end
+    else
+    flash[:notice] = "You do not have permission to edit that tweet."
+    redirect '/login'
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      @user = @tweet.user
+      erb :'/tweets/show'
+    else
+      redirect '/login'
+    end
+  end
+
+  patch '/tweets/:id' do
+    if params[:content] == ""
+      redirect "/tweets/#{params[:id]}/edit"
+    else
+      @tweet = Tweet.find(params[:id])
+      @tweet.update(content: params[:content])
+    end
+    redirect "/tweets/#{@tweet.id}"
+  end
+
+  delete '/tweets/:id/delete' do
+  if logged_in? && current_user.tweets.find_by(id: params[:id])
+    @tweet = current_user.tweets.find_by(id: params[:id])
+    if @tweet && @tweet.destroy
+      @tweet.destroy
+      redirect "/tweets"
+    else
+      redirect "/tweets/#{tweet.id}"
+    end
+  end
+  end
+
+  post '/tweets' do
+    if logged_in?
+      @tweet = Tweet.new(content: params[:content], user_id: session[:id])
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
@tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.